bootq/src/main/java/life/genny/bootq/models/BatchLoading.java

```diff
@@ -267,15 +292,36 @@ public void persistDefBaseEntityAttributes(Map<String, Map<String, String>> proj
     public void persistBaseEntityAttributes(Map<String, Map<String, String>> project, String realmName) {
         Instant start = Instant.now();
         for (Map.Entry<String, Map<String, String>> entry : project.entrySet()) {
-			try {
-				EntityAttribute entityAttribute = googleSheetBuilder.buildEntityAttribute(entry.getValue(), realmName);
-				beaUtils.updateEntityAttribute(entityAttribute);
-			} catch (ItemNotFoundException e) { // ensure to print beCode and eaCode
-				log.warn(new StringBuilder("Error occurred when building ")
-                    .append(entry.getValue().get("baseentitycode")).append(":").append(entry.getValue().get("attributecode"))
+			
+            String baseEntityCode = entry.getValue().get("baseentitycode");
+            String attributeCode = entry.getValue().get("attributecode");
+
+            String combined = new StringBuilder(baseEntityCode).append(":").append(attributeCode).toString();
+
+            EntityAttribute entityAttribute;
+            try {
+                log.trace("Building " + combined + " entityAttribute");
+                entityAttribute = googleSheetBuilder.buildEntityAttribute(entry.getValue(), realmName);
+                // Attribute ID fix (post build, this is where the responsibility lies)
+                if(entityAttribute.getAttributeId() == null) {
+                    Long maxId = cm.getMaxAttributeId();
+                    log.error("Detected null attribute id for: " + combined + ". Setting to: " + maxId);
+                    entityAttribute.setAttributeId(maxId);
```

Comment:
A couple of points here:
- This should be setting maxId + 1
- This should be done inside the GoogleSheeetBuilder method for consistency sake.

Reply:
Solid point, to make this work properly though it should be done as maxId + 1 and when attributes are created, not entity attributes. This will cause detachment as the existing attributes won't have IDs otherwise. This exists already but also needs to be amended. This line needs to be removed